src/main/java/org/opensearch/knn/index/codec/nativeindex/MemOptimizedNativeIndexBuildStrategy.java

```diff
@@ -52,7 +52,8 @@ public static MemOptimizedNativeIndexBuildStrategy getInstance() {
      * @param knnVectorValues  The {@link KNNVectorValues} representing the vectors to be indexed.
      * @throws IOException     If an I/O error occurs during the process of building and writing the index.
      */
-    public void buildAndWriteIndex(final BuildIndexParams indexInfo, final KNNVectorValues<?> knnVectorValues) throws IOException {
+    public void buildAndWriteIndex(final BuildIndexParams indexInfo) throws IOException {
```

Comment:
This was purposefully kept different, data to be indexed is not an index parameter. Its best to keep them logically independent even in the code, I would rather be okay with a 3rd parameter then bloating BuildIndexParams

Reply:
Sure let me handle this in a refactoring PR. and merging this change to ensure that we are meeting the code freeze deadline.